libp2p_test.go

```diff
@@ -39,6 +40,39 @@ func TestInsecure(t *testing.T) {
 	h.Close()
 }
 
+func TestDefaultListenAddrs(t *testing.T) {
```

Comment:
Should have a "correctly set the default listeners" test (i.e., make sure this actually works).

Reply:
Something like this?
```go
h, err := New(ctx)
if err != nil {
	t.Fatal(err)
}
if len(h.Addrs()) != 2 {
	t.Error("expected 2 default listen addrs")
}
h.Close()
```

Is length of ```h.Addrs()``` sufficient for tests or should they be compared with the actual strings?

Reply:
That's probably good enough. Thanks!

Reply:
I think that test error is due to how we expand ALL addresses using the interface addresses in `Host.Addrs()`. Try calling `h.Network().ListenAddresses()`. In that case, we should even be able to verify that they are the correct addresses.